File: LicenseManagementApi/Models/Entities/RsaKey.cs
namespace LicenseManagementApi.Models.Entities;

public class RsaKey
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string PublicKey { get; set; } = string.Empty;
    public string PrivateKeyEncrypted { get; set; } = string.Empty;
    public int KeySize { get; set; }
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public ICollection<License> Licenses { get; set; } = new List<License>();
}

File: vue-ui-frontend/src/types/rsaKey.ts
export interface RsaKey {
  id: string
  name: string
  publicKey: string
  keySize: number
  createdBy: string
  createdAt: string
  updatedAt: string
}

export interface CreateRsaKeyDto {
  name: string
  keySize: number
  createdBy: string
}
